client/src/holdings/AddTradeDialog.tsx

```diff
@@ -30,6 +46,35 @@ export function AddTradeDialog({
   showNotification,
 }: AddTradeDialogProps) {
   const [open, setOpen] = useState(false);
+  const [selectedDate, setSelectedDate] = useState(new Date());
+  const handleDateChange = (date: any) => {
+    setSelectedDate(date);
+  };
+  const tradeValidationSchema = yup.object().shape({
+    category: yup.string(),
+    date: yup.date(),
+    broker: yup.string(),
+    quantity: yup.number().required().moreThan(0),
+    priceCurrency: yup.string(),
+    unitPrice: yup.number().required().moreThan(0),
+    fee: yup.string(),
+    tax: yup.string(),
+    fxRate: yup.string(),
+  });
+  const {
+    register,
+    handleSubmit,
+    control,
+    errors,
+    formState: {isSubmitting},
+  } = useForm<CreateTradeData>({
+    resolver: yupResolver(tradeValidationSchema),
+    defaultValues: {
+      category: 'BUY',
```

Comment:
You need to add a date default value here to work with my other comments on the date

`date: new Date().toISOString(),`